src/main/java/macrobase/analysis/pipeline/OneShotPipeline.java

```diff
@@ -0,0 +1,36 @@
+package macrobase.analysis.pipeline;
+
+import macrobase.analysis.result.AnalysisResult;
+import macrobase.conf.ConfigurationException;
+import macrobase.conf.MacroBaseConf;
+
+import java.io.IOException;
+import java.sql.SQLException;
+
+
+abstract public class OneShotPipeline extends AbstractPipeline {
```

Comment:
I wonder if we could have the final stage of the pipeline be something that doesn't sub-class Iterator, but accumulates results instead into some internal state. Then we could support multiple functions on it such as `getNewResults`, `getAllResults`, etc. Seems more malleable then the iterator's `next` approach.